audmath/core/api.py

```diff
@@ -86,6 +86,183 @@ def db(
+            If provided as string without unit,
+            e.g. ``'2000'``,
+            it is treated as samples
+            and will be converted with the help of ``sampling_rate``
+            to seconds
+        sampling_rate: sampling rate in Hz.
+            Has to be provided
+            if time is provided in samples
+
+    Returns:
+        time in seconds
+
+    Raises:
+        ValueError: if ``time`` is provided in samples
+            but ``sampling_rate`` is ``None``
+        ValueError: if the provided unit is not supported
+
+    Examples:
+        >>> duration_in_seconds(2)
+        2.0
+        >>> duration_in_seconds(2.0)
+        2.0
+        >>> duration_in_seconds('2ms')
+        0.002
+        >>> duration_in_seconds('ms')
+        0.001
+        >>> duration_in_seconds('2000', 1000)
+        2.0
+        >>> duration_in_seconds(np.timedelta64(2, 's'))
+        2.0
+
+    """  # noqa: E501
+    # Dictionary with allowed unti entries
+    # and mapping from pandas.to_timedelta()
+    # to numpy.timedelta64, see
+    # https://numpy.org/doc/stable/reference/arrays.datetime.html#datetime-units
+    unit_mapping = {
+        # week
+        'W': 'W',
+        # day
+        'D': 'D',
+        'days': 'D',
+        'day': 'D',
+        # hour
+        'h': 'h',
+        'hours': 'h',
+        'hour': 'h',
+        'hr': 'h',
+        # minute
+        'm': 'm',
+        'minutes': 'm',
+        'minute': 'm',
+        'min': 'm',
+        'T': 'm',
+        # second
+        's': 's',
+        'seconds': 's',
+        'second': 's',
+        'sec': 's',
+        'S': 's',
+        # millisecond
+        'ms': 'ms',
+        'milliseconds': 'ms',
+        'millisecond': 'ms',
+        'millis': 'ms',
+        'milli': 'ms',
+        'L': 'ms',
+        # microsecond
+        'us': 'us',
+        'μs': 'us',
+        'microseconds': 'us',
+        'microsecond': 'us',
+        'micros': 'us',
+        'micro': 'us',
+        'U': 'us',
+        # nanosecond
+        'ns': 'ns',
+        'nanoseconds': 'ns',
+        'nanosecond': 'ns',
+        'nanos': 'ns',
+        'nano': 'ns',
+        'N': 'ns',
+    }
+
+    # numpy.timedelta64() accepts only integer as input values,
+    # so we need to convert all values
+    # to nanoseconds and integers first
+    def to_nanos(value, unit):
+        if unit == 'W':
+            value = value * 7 * 24 * 60 * 60 * 10 ** 9
+        elif unit == 'D':
+            value = value * 24 * 60 * 60 * 10 ** 9
+        elif unit == 'h':
+            value = value * 60 * 60 * 10 ** 9
+        elif unit == 'm':
+            value = value * 60 * 10 ** 9
+        elif unit == 's':
+            value = value * 10 ** 9
+        elif unit == 'ms':
+            value = value * 10 ** 6
+        elif unit == 'us':
+            value = value * 10 ** 3
+        return int(value)
+
+    if isinstance(time, str):
+
+        # ensure we have a str and not numpy.str_
+        time = str(time)
```

Comment:
```suggestion
        time = str(time).replace(' ', '')
```

Then something like "10 ms" is also supported.

Reply:
Very nice idea.

I decided to implement it differently, to raise meaningful errors if a user provides `'10 0 ms'` or other typos
and updated the description of this pull request.

Reply:
But now the following can happen:

```python
audmath.duration_in_seconds('3s ')
```
```
ValueError: could not convert string to float: '3s'
```

So maybe at least change to:

```python
        duration = str(duration).strip()
```

Reply:
I handled spaces at the beginning and end, and raise a new custom error if the string is empty or contains only a single space:

```python
>>> audmath.duration_in_seconds('3s ')
3.0
>>> audmath.duration_in_seconds(' 3s')
3.0
>>> # new error message when providing only spaces
>>> audmath.duration_in_seconds('   ')
...
ValueError: 'duration' is not allowed to be empty, or contain only spaces.
>>> # or empty string
>>> audmath.duration_in_seconds('')
...
ValueError: 'duration' is not allowed to be empty, or contain only spaces.
```

Reply:
This has now changed to:

```python
>>> audmath.duration_in_seconds('3s ')
3.0
>>> audmath.duration_in_seconds(' 3s')
3.0
>>> audmath.duration_in_seconds('   ')
...
ValueError: Your given duration '' is not conform to the 'value' 'unit' pattern.
>>> audmath.duration_in_seconds('')
...
ValueError: Your given duration '' is not conform to the 'value' 'unit' pattern.
```

Reply:
As I found it a little bit confusing for the user that we present a different duration value in the error message than the one provided by the user, I replaced `.strip()` by matching spaces at the beginning and end of the string directly in the compile pattern. This means we are now getting here:

```python
>>> audmath.duration_in_seconds('   ')
...
ValueError: Your given duration '   ' is not conform to the 'value' 'unit' pattern.
>>> audmath.duration_in_seconds('')
...
ValueError: Your given duration '' is not conform to the 'value' 'unit' pattern.
```